agent/consul/state/catalog.go

```diff
@@ -4016,6 +4016,7 @@ func insertGatewayServiceTopologyMapping(tx WriteTxn, idx uint64, gs *structs.Ga
 	mapping := upstreamDownstream{
 		Upstream:   gs.Service,
 		Downstream: gs.Gateway,
+		Refs:       make(map[string]struct{}),
```

Comment:
We only create this `upstreamDownstream` type in two places. The other place already properly initializes the `Refs`, so this seemed like the most direct fix.